library/std/src/os/uefi/env.rs

```diff
@@ -0,0 +1,47 @@
+//! UEFI-specific extensions to the primitives in `std::env` module
+
+use super::raw::{BootServices, RuntimeServices, SystemTable};
+use crate::ffi::c_void;
+use crate::ptr::NonNull;
+use crate::sync::atomic::{AtomicPtr, Ordering};
+
+static GLOBAL_SYSTEM_TABLE: AtomicPtr<SystemTable> = AtomicPtr::new(crate::ptr::null_mut());
+static GLOBAL_SYSTEM_HANDLE: AtomicPtr<c_void> = AtomicPtr::new(crate::ptr::null_mut());
+
+/// Initializes Global Atomic Pointers to SystemTable and Handle.
+/// Should only be called once in the program execution under normal circumstances.
+/// The caller should ensure that the pointers are valid.
+#[unstable(feature = "uefi_std", issue = "100499")]
+pub fn init_globals(handle: NonNull<c_void>, system_table: NonNull<SystemTable>) {
+    // Use `Relaxed` ordering since there is only one thread
+    GLOBAL_SYSTEM_TABLE.store(system_table.as_ptr(), Ordering::Relaxed);
+    GLOBAL_SYSTEM_HANDLE.store(handle.as_ptr(), Ordering::Relaxed);
```

Comment:
These stores both need to be Release, not Relaxed.

Reply:
There is no observable difference when you only have a single thread.

Reply:
That's not true, the compiler can re-order other stores across relaxed.

Reply:
But not in a way that changes observable semantics in a single thread, IIUC.

Reply:
In principal we could compiler_fence to avoid that (And in this case it likely does not matter because the stores into the memory behind these pointers presumably happen outside of rust), but it's cleaner and less error-prone just to use the correct orderings.